dftimewolf/lib/collectors/grr_hosts.py

```diff
@@ -156,15 +156,18 @@ def _await_flow(self, client, flow_id):
         msg = 'Unable to stat flow {0:s} for host {1:s}'.format(
             flow_id, client.data.os_info.fqdn.lower())
         self.state.add_error(msg)
-        raise RuntimeError(
+        raise DFTimewolfError(
             'Unable to stat flow {0:s} for host {1:s}'.format(
                 flow_id, client.data.os_info.fqdn.lower()))
 
       if status.state == flows_pb2.FlowContext.ERROR:
         # TODO(jbn): If one artifact fails, what happens? Test.
-        raise RuntimeError(
-            '{0:s}: FAILED! Backtrace from GRR:\n\n{1:s}'.format(
-                flow_id, status.context.backtrace))
+        message = status.context.backtrace
+        if 'ArtifactNotRegisteredError' in status.context.backtrace:
```

Comment:
AFAIK all GRR provides is a string, so we unfortunately need to search through it to get more details on the error.